astroid/brain/brain_typing.py

```diff
@@ -196,6 +196,28 @@ def infer_typing_attr(
     return node.infer(context=ctx)
 
 
+def _looks_like_generic_class_pep695(node: ClassDef) -> bool:
+    """Check if class is uses type parameter. Python 3.12+."""
+    return len(node.type_params) > 0
+
+
+def infer_typing_generic_class_pep695(
+    node: ClassDef, ctx: context.InferenceContext | None = None
+) -> Iterator[ClassDef]:
+    """Add __class_getitem__ for generic classes. Python 3.12+."""
+    try:
+        value = next(node.infer())
```

Comment:
Thanks for tackling this. One question for my own information: why do we need to use inference? Can we just check `node.type_params`?

Reply:
Tbh I'm not quite sure either. All other methods do this as well. My best guess is that with the `brain_**` we overwrite the default inference. However, since we still want that, we need to call `node.infer()` ourselves.

Reply:
Interestingly, if I add `assert node is value`, it passes.

Can we use this opportunity to explore avoiding inference here?